src/libratbag.c

```diff
@@ -196,6 +196,11 @@ ratbag_device_new(struct ratbag *ratbag, struct udev_device *udev_device,
 	device->udev_device = udev_device_ref(udev_device);
 	device->ids = *id;
 	device->data = ratbag_device_data_new_for_id(ratbag, id);
+
+	if (device->data != NULL) {
+		device->device_type = (char*)ratbag_device_data_get_device_type(device->data);
```

Comment:
There are very few reasons why you'd want to cast a `const char*` into a `char*` and I don't think any of them apply here. As a general rule, if you need to do this, something isn't right.